src/chapters/databases/db-tips.rst

```diff
@@ -0,0 +1,325 @@
+
+   #. Open ``main.py`` and create the new database file. Its name must match
+      the one in ``.gitignore``.
+
+      .. sourcecode:: python
+         :linenos:
+
+         import sqlite3
+
+         database = sqlite3.connect('no_gitting_this.db')
+
+   #. Run ``main.py``. Notice that when the new database appears in the file
+      tree, its name remains gray.
+
+      .. figure:: figures/db-ignore.png
+         :alt: Showing the file tree with database names greyed out.
+
+   #. Run ``git status`` in the terminal. Notice that ``main.py`` and
+      ``.gitignore`` appear in the output, but not the new database.
+
+      .. sourcecode:: bash
+
+         $ git status
+         On branch main
+         Changes not staged for commit:
+         (use "git add <file>..." to update what will be committed)
+
+            modified:   .gitignore
+            modified:   main.py
+
+Add a Primary Key
+-----------------
+
+.. index:: ! primary key
+
+To help mange the data stored in a table, one helpful tool is the
+**primary key**. This is an integer assigned to each row in the table, and no
+two rows have the same primary key.
+
+By adding primary keys, every row in the table will be different from every
+other row. For example, if we have two students with the same name, their table
+entries can easily be confused. However, by adding a unique primary key, the two
+rows no longer look the same.
+
+.. admonition:: Example
+
+   Without primary keys, we cannot tell the difference between two rows that
+   contain identical values. Primary keys guarantee that every row has at least
+   one unique piece of data.
+
+   .. figure:: figures/primary-key.png
+      :alt: Showing two database tables. One with a primary key column and the other without.
+      :width: 80%
+
+      The table on the right includes an ``id`` column that stores the primary key for each row.
+
+When we use the ``sqlite3`` module, we can assign primary keys automatically.
+However, we must include some special syntax when we first create the table.
+
+.. sourcecode:: SQL
+
+   CREATE TABLE table_name (key_name INTEGER PRIMARY KEY, other_columns...)
+
+#. ``key_name`` is the column in the table that will store the primary keys.
+   Each key value must be an integer data type.
+#. We must type out ``INTEGER`` instead of ``INT`` for the data type. This is
+   required! Using ``INT`` causes errors.
+#. ``PRIMARY KEY`` sets up the process to automatically assign an integer to
+   ``key_name``. This happens each time a new row is added to the table. We do
+   NOT need to assign a value to ``key_name`` ourselves.
+#. Python keeps track of the largest primary key in the table. When a new row
+   is added, its primary key is assigned the next higher value.
+#. We can assign our own value to ``key_name`` when we ``INSERT`` a row.
+   However, the action will throw an error if the value we choose matches a
+   primary key already in the table.
+
```

Comment:
Not sure if you want to mention that you can add a primary key to an existing table, its just more complicated.

Reply:
I did consider this when I was writing the page, but I thought that keeping things simple was a better option.

Starting with a PK column is the practice we want to enforce.  If motivated students want to change an existing table, I'm betting they can research it themselves.  ;-)

BTW, `sqlite3` automatically assigns primary keys, even if there's no column in the table for them.  However, that is definitely a rabbit hole I didn't want the kids to follow me down!

Reply:
I might have to venture down that rabbit hole... thanks for the tip!